addons/mass_mailing/models/mail_mail.py

```diff
@@ -26,8 +28,15 @@ def create(self, values_list):
 
     def _get_tracking_url(self):
         base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
-        track_url = werkzeug.urls.url_join(base_url, 'mail/track/%s/blank.gif' % self.id)
-        return '<img src="%s" alt=""/>' % track_url
+        token = self._get_tracking_token()
+        return werkzeug.urls.url_join(base_url, f'mail/track/{self.id}/blank.gif?token={token}')
+
+    def _get_tracking_token(self):
```

Comment:
Could you move that to tools (and/or check we don't have existing tool) so that we avoid recreating the "roughly the same" code every time we need a token ? Thanks!